service/history/shard/context_impl.go

```diff
@@ -1750,10 +1747,6 @@ func (s *ContextImpl) acquireShard(newMaxReadLevel int64) {
 			s.engine = engine
 		}
 
-		// Set max read level after a re-acquisition (if this is the first
-		// acquisition, newMaxReadLevel will be zero so it's a no-op)
-		s.updateMaxReadLevelLocked(newMaxReadLevel)
```

Comment:
https://github.com/temporalio/temporal/pull/2724/files#diff-428a9306aaf7598228061976d6e37593a7be02c0bb45449091e5efaaa9f08edbR1725

this line already updated the max readable task ID (due to range ID increase)